userdocs/diagnostics/expression-too-complex.md

```diff
@@ -0,0 +1,16 @@
+# Type Inference in Complex Expressions
+For some complex expressions, the Swift compiler is unable to infer types in reasonable time. This can occur for a number of reasons, but some common ones include:
+
+- The expression makes use of many overloaded operators and/or functions.
+- The expression contains literals which can be used as values of many different types (e.g. integer and floating point literals).
+- The expression includes a collection literal with a large number of nontrivial element expressions.
+
+These types of expressions can take a long time to typecheck because the compiler must consider and eliminate a larger number of possible solutions when assigning types to subexpressions.
+
+When this error occurs, possible solutions include:
+
+- Breaking the expression up into multiple subexpressions by storing intermediate values as `let` constants. These simpler subexpressions can have their types inferred more quickly.
```

Comment:
Nit: The juxtaposition of two prepositions ("up into") can be difficult to parse for non-native English speakers, particularly since "breaking up" is a more informal expression with several possible dictionary definitions. Consider instead "Breaking up the expression into..." or perhaps more simply, "Separating the expression into..."